test/unit/render-util.test.js

```diff
@@ -16,6 +19,46 @@ describe('core/render/utils', () => {
       expect(result).toEqual('content');
     });
   });
+
+  // getAndRemoveConfig()
+  // ---------------------------------------------------------------------------
+  describe('getAndRemoveConfig()', () => {
+    test('parse simple config', () => {
+      const result = getAndRemoveConfig(
+        `[filename](_media/example.md ':include')`
+      );
+
+      expect(result).toMatchObject({
+        config: {},
+        str: `[filename](_media/example.md ':include')`,
+      });
+    });
+
+    test('parse config with arguments', () => {
+      const result = getAndRemoveConfig(
+        `[filename](_media/example.md ':include :foo=bar :baz test')`
+      );
+
+      expect(result).toMatchObject({
+        config: {
+          foo: 'bar',
+          baz: true,
+        },
+        str: `[filename](_media/example.md ':include test')`,
+      });
+    });
+
+    test('parse config with double quotes', () => {
+      const result = getAndRemoveConfig(
+        `[filename](_media/example.md ":include")`
+      );
+
+      expect(result).toMatchObject({
+        config: {},
+        str: `[filename](_media/example.md ":include")`,
+      });
+    });
+  });
```

Comment:
I wonder what the behavior should actually be. Seems funky that it leaves things behind. Seems like it should handle everything.